lib/src/connection.dart

```diff
@@ -97,6 +97,12 @@ abstract class Connection {
   /// Active state handler for this connection.
   ActiveStateHandler onActiveStateChanged;
 
+  final Completer<void> _onConnectionOperational = Completer<void>();
+
+  /// Future which completes when connection becomes operational after
+  /// we receive the initial SETTINGS frame from the other side.
+  Future<void> get onConnectionOperational => _onConnectionOperational.future;
```

Comment:
IIRC The connection is operational right from the start. One can start making streams right from the start.

Maybe rename this to something like `onInitialPeerSettingsReceived`.

Reply:
I used (internal) terminology from http2 package itself. But I agree that I should rename it.